responsibleai/tests/counterfactual_manager_validator.py

```diff
@@ -22,9 +22,16 @@ def verify_counterfactual_object(counterfactual_obj, feature_importance=False):
 
 
 def validate_counterfactual(cf_analyzer, X_train, target_column,
+                            categorical_features=None,
                             desired_class=None, desired_range=None,
                             feature_importance=False):
-    continuous_features = list(set(X_train.columns) - set([target_column]))
+
+    if categorical_features is None:
```

Comment:
This is fine, but another way to do this would be:
```
if categorical_features is None:
    categorical_features = set()

continuous features = ...
```
You still need to test both branches, but you duplicate less logic

Reply:
The tests are for both cases. Don't want to create an empty set if not needed.

Reply:
Sure, you're trading off the small memory overhead of creating an empty set vs a small amount of code duplication. Probably just a matter of preference